### PR TITLE
Blog: How to Use If-Else Logic in Node-RED: A Step-by-Step Guide

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by ZJvandeWeg** [comment]
@sumitshinde-84 "Design Patterns" I think is a bit much, for example; the Gang of 4 in computer science are patterns of code structure. See also: https://en.wikipedia.org/wiki/Design_Patterns

Loops, if-else, etc, etc are more fundamental, so maybe "Flow Programming Fundamentals"?

File: src/blog/2024/08/node-red-design-patterns.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
+title: "Building Better Node-RED Flows: A Comprehensive Guide to Design Patterns and Best Practices"
```
Comment (ZJvandeWeg) [inline src/blog/2024/08/node-red-design-patterns.md:2]: Let's scope this down and have 600 words just about the following? Keep the rest in our pocket for further articles?
```suggestion
title: "Building Better Node-RED Flows: If Else statements and conditional logic"
```

File: src/blog/2024/09/node-red-if-else-logic-guide.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+This approach allows systems to respond appropriately to different situations.
+
+## Implementing Conditional Flows in Node-RED: A Practical Walkthrough
+
+In Node-RED, implementing If-Else logic allows you to create dynamic and responsive flows that react to different inputs and conditions. Whether you're automating a smart home, managing IoT devices, or developing complex workflows, mastering conditional logic is essential for creating intelligent systems.
+
+To implement If-Else logic in Node-RED, you can use the Function node. However, Node-RED also offers a low-code approach, and there’s another popular method to implement conditional logic without writing code, which we will also cover in this article.
+
+### Using Function Node
```
Comment (ZJvandeWeg) [inline src/blog/2024/09/node-red-if-else-logic-guide.md:50]: Let's start with the switch node

Comment (sumitshinde-84) [reply src/blog/2024/09/node-red-if-else-logic-guide.md:50]: But a switch is not the if-else , in the context of programming.

File: src/blog/2024/09/images/if-else-advance.png
Comment (ZJvandeWeg) [inline src/blog/2024/09/images/if-else-advance.png:1]: Should be removed?

File: src/blog/2024/09/images/if-else-function-1.png
Comment (ZJvandeWeg) [inline src/blog/2024/09/images/if-else-function-1.png:1]: Can we remove this?

File: src/blog/2024/09/images/if-else-switch.png
Comment (ZJvandeWeg) [inline src/blog/2024/09/images/if-else-switch.png:1]: This has been rendered in page, and we can remove this from the PR / Git branch?

File: src/blog/2024/09/node-red-if-else-logic-guide.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    - Rule 1: `msg.payload > 20` 
+    - Rule 2: `msg.payload > 10` 
+
+5. Now drag the Debug nodes and connect them to the Switch nodes' outputs according to our example. For messages greater than 30, connect the Debug node to the first output of the first Switch node. For the range between 30 to 20, connect the Debug node to the first output of the second Switch node. For the range between 20 to 10, connect the Debug node to the second output of the second Switch node. Finally, for messages less than 10, connect the Debug node to the fourth output of the first Switch node.
+6. Deploy the flow by clicking the "Deploy" button in the top-right corner of the Node-RED editor.
+7. Once deployed, click the button on the Inject node to trigger it. The Debug nodes will show the routed messages based on the temperature value.
+
+Notice how messages are routed through different outputs based on the temperature value. Now, you may ask how to update the message payload based on a condition. For that, you will need to use the Change node or the Function node.
+
+```mermaid
```
Comment (ZJvandeWeg) [inline src/blog/2024/09/node-red-if-else-logic-guide.md:76]: @Yndira-E Can you check why these aren't rendered? I think the library needs loading?

Comment (sumitshinde-84) [reply src/blog/2024/09/node-red-if-else-logic-guide.md:76]: Maybe it is not rendering in the Netlify preview.

Comment (Yndira-E) [reply src/blog/2024/09/node-red-if-else-logic-guide.md:76]: You were right, ZJ. I've added the Mermaid script to the posts template and now the diagrams are rendering correctly.

**Review by ZJvandeWeg — APPROVED** [review]

**Pull request merged**